src/bridge/cockpitpackages.c

```diff
@@ -44,10 +45,63 @@ const gchar **cockpit_bridge_data_dirs = NULL; /* default */
 
 gint cockpit_bridge_packages_port = 0;
 
+/* Packages might change while the bridge is running, and we support
+   that with slightly complicated handling of checksums.
+
+   The bridge reports a single checksum for the whole bundle of
+   packages.  This is the checksum that ends up in URLs and cockpit-ws
+   makes routing decisions based on it.
+
+   When the packages change on disk, this bundle checksum also
+   changes.  However, the bridge will not change what it reports; it
+   will keep reporting the original bundle checksum.  This ensures
+   that URLs that use the original checksum continue to work.
+
+   The manifest for a package also contains a checksum, and this
+   checksum will change when the package changes.  The shell can use
+   this second checksum to decide whether to reload a component, for
+   example.
+
+   The checksum of a package in its manifest is also a bundle
+   checksum.  More precisely, it is the oldest bundle checksum that
+   the bridge has seen that includes the exact files of the given
+   package.
+
+   Thus, after the bridge has started, the reported checksum and all
+   manifest checksums are the same.  If a new package appears but none
+   of the old packages are changed, the new package has the new bundle
+   checksum in its manifest, and all the old packages still have the
+   reported checksum.
+
+   In order to load the files of a new package, the shell should not
+   use the reported bridge checksum.  Loading will succeed, but the
```

Comment:
This is incorrect: Loading might succeed, but it might also fails if cockpit-ws routes the request to a different host.